### PR TITLE
split travis into e2e_tests and unit_tests

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/bazel/issues/103

### CODE REVIEW

File: tool/presubmit.sh [hunk 1]
```diff
@@ -3,19 +3,47 @@
...
-pub upgrade
-dartanalyzer --fatal-warnings lib/_bazel_codegen.dart
-pub run test
-popd
+e2e_tests () {
+  if [ "$TRAVIS" == "true" ]; then
+    echo "Installing bazel"
+    # Install bazel if we are on travis.
+    wget https://github.com/bazelbuild/bazel/releases/download/0.4.4/bazel_0.4.4-linux-x86_64.deb
+    sudo dpkg -i bazel_0.4.4-linux-x86_64.deb
```
Comment (natebosch) [inline tool/presubmit.sh]: This doesn't work with Travis containers. I think we can add some config to get around it but I'm not sure it's a good idea.

Was the goal to get faster feedback from unit tests?

File: .travis.yml [hunk 2]
```diff
@@ -12,8 +12,11 @@ cache:
     - $HOME/.pub-cache
 
 before_install:
-  - wget https://github.com/bazelbuild/bazel/releases/download/0.4.4/bazel_0.4.4-linux-x86_64.deb
-  - sudo dpkg -i bazel_0.4.4-linux-x86_64.deb
+  - if [ "$TRAVIS" == "true" ]; then wget https://github.com/bazelbuild/bazel/releases/download/0.4.4/bazel_0.4.4-linux-x86_64.deb; sudo dpkg -i bazel_0.4.4-linux-x86_64.deb; fi
```
Comment (natebosch) [inline .travis.yml]: when would this be run such that $TRAVIS is not "true"?

Comment (jakemac53) [inline .travis.yml]: if you are running the tests locally

Comment (natebosch) [inline .travis.yml]: I've never had .travis.yml as a factor when I run tests locally...

Comment (jakemac53) [inline .travis.yml]: oh ya this was in the shell script but now it isn't necessary, obviously :)

**Pull request merged**